css-shared-element-transitions/index.bs

```diff
@@ -593,254 +655,307 @@ document's DOM structure.
+
+    1. If |document|'s [=document/running same document transition=] is |transition|,
+        then set |document|'s [=document/running same document transition=] to null.
+
+    1. [=Reject=] |transition|'s [=SameDocumentTransition/ready promise=] with |error|.
+
+    1. [=Reject=] |transition|'s [=SameDocumentTransition/finished promise=] with |error|.
 </div>
 
 <div algorithm>
-For <dfn>capturing the image</dfn> of an {{Element}} |el|:
-
-	1. Render the referenced element and its descendants,
-		at the same size that they would be in the document,
-		over an infinite transparent canvas with the following
-		characterists:
-	
-		1. The origin of |el|'s
-			<a href="https://drafts.csswg.org/css-overflow/#ink-overflow-rectangle">ink overflow rectangle</a>
-			is anchored to canvas origin.
-	
-		1. If the referenced element has a transform applied to it (or its ancestors),
-			the transform is ignored.
-	
-			Note: This transform is applied to the snapshot using the `transform`
-				property of the associated [=page-transition-container=] pseudo-element.
-	
-		1. For each descendant with a non-''page-transition-tag/none'' [=page-transition-tag=],
-			  skip painting this descendant.
-	
-			Note: This is necessary since the descendant will generate its
-				own snapshot which will be displayed and animated independently.
-
-	1. Let |interest rectangle| be the result of [=Compute the interest rectangle=] for |el|.
-		The |interest rectangle| is the subset of |el|'s
-		<a href="https://drafts.csswg.org/css-overflow/#ink-overflow-rectangle">ink overflow rectangle</a>
-		that should be captured.
-	
-		Note: This is required for cases where an element's ink overflow rectangle needs
-			to be clipped because of hardware constraints. For example, if it exceeds
-			max texture size.
-	
-	1. Set the captured image to the contents of the |interest rectangle| in the rendered canvas.
-		The natural size of the image is equal to the |interest rectangle| bounds.
+    To <dfn lt="capture the image|capturing the image">capture the image</dfn> given an {{Element}} |element|, perform the following steps.
+    They return an image.
+
+    1. Render the referenced element and its descendants,
+        at the same size that they would be in the document,
+        over an infinite transparent canvas with the following characteristics:
+
+        * The origin of |element|'s [=ink overflow rectangle=] is anchored to canvas origin.
+
+        * If the referenced element has a transform applied to it (or its ancestors),
+            then the transform is ignored.
+
+            Note: This transform is applied to the snapshot using the `transform` property of the associated ''::page-transition-container'' pseudo-element.
+
+        * [=list/For each=] |descendant| of [=shadow-including descendant=] {{Element}} and [=pseudo-element=] of |element|,
+            if |descendant| has a [=computed value=] of 'page-transition-tag' that is not ''page-transition-tag/none'',
```

Comment:
For a moment I thought this might break in the case where an element with a non-none computed value doesn't participate in the transition. But that only happens if its not-renedered, in which case this would be a no-op. Maybe worth adding a note to clarify this.

Reply:
Hmm yeah, this duplicates the logic for deciding if an element should participate in the transition. I've added a TODO to fix this.